src/main/java/org/gridsuite/modification/server/modifications/ByFilterDeletion.java

```diff
@@ -0,0 +1,107 @@
+        var filters = modificationInfos.getFilters().stream()
+                .filter(distinctByKey(FilterInfos::getId))
+                .collect(Collectors.toMap(FilterInfos::getId, FilterInfos::getName));
+
+        Map<UUID, FilterEquipments> exportFilters = ModificationUtils.getUuidFilterEquipmentsMap(filterService, network, subReporter, filters, modificationInfos.getErrorType());
+        if (exportFilters != null) {
+            Map<UUID, FilterEquipments> exportedFiltersWithWrongEquipmentIds = ModificationUtils.getUuidFilterWrongEquipmentsIdsMap(subReporter, exportFilters, filters);
+            List<IdentifiableAttributes> identifiableAttributes = ModificationUtils.getIdentifiableAttributes(exportFilters, exportedFiltersWithWrongEquipmentIds, modificationInfos.getFilters(), subReporter);
+
+            if (CollectionUtils.isEmpty(identifiableAttributes)) {
+                String filterNames = modificationInfos.getFilters().stream().map(FilterInfos::getName).collect(Collectors.joining(", "));
+                createReport(subReporter,
+                        "allFiltersWrong",
+                        String.format("All of the following filters have equipments with wrong id : %s", filterNames),
+                        TypedValue.WARN_SEVERITY);
+            } else {
+                applyFilterDeletion(network, subReporter, identifiableAttributes);
+            }
+
+            subReporter.report(Report.builder()
+                    .withKey("equipmentDeleted")
+                    .withDefaultMessage("equipment of type=${type} and ids=${ids} deleted")
+                    .withValue("type", modificationInfos.getEquipmentType().name())
+                    .withValue("ids", identifiableAttributes.stream().map(IdentifiableAttributes::getId).collect(Collectors.joining(", ")))
+                    .withSeverity(TypedValue.INFO_SEVERITY)
+                    .build());
+        }
+    }
+
+    private void applyFilterDeletion(Network network, Reporter subReporter, List<IdentifiableAttributes> identifiableAttributes) {
+        IdentifiableType identifiableType = modificationInfos.getEquipmentType();
+        if (CONNECTABLE_EQUIPMENTS.contains(identifiableType)) {
+            identifiableAttributes.forEach(identifiableAttribute -> new RemoveFeederBay(identifiableAttribute.getId()).apply(network, true, subReporter));
+        } else if (identifiableType == IdentifiableType.VOLTAGE_LEVEL) {
+            identifiableAttributes.forEach(identifiableAttribute -> new RemoveVoltageLevel(identifiableAttribute.getId()).apply(network, true, subReporter));
+        } else if (identifiableType == IdentifiableType.SUBSTATION) {
+            identifiableAttributes.forEach(identifiableAttribute -> new RemoveSubstationBuilder().withSubstationId(identifiableAttribute.getId()).build().apply(network, true, subReporter));
+        } else {
+            throw new PowsyblException("Unsupported equipment type");
```

Comment:
Throws a `NetworkModificationException` because it's not a powsybl problem
`createEquipmentTypeUnsupported()`